sdk/src/main/java/com/hedera/hashgraph/sdk/HederaTrustManager.java

```diff
@@ -0,0 +1,63 @@
+package com.hedera.hashgraph.sdk;
+
+import org.bouncycastle.crypto.digests.SHA384Digest;
+import org.bouncycastle.util.encoders.Hex;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.X509TrustManager;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+class HederaTrustManager implements X509TrustManager {
+    @Nullable
+    private final String certHash;
+
+    HederaTrustManager(@Nullable NetworkName networkName, AccountId nodeAccountId) {
+        if (networkName == null) {
+            this.certHash = null;
+            return;
+        }
+
+        var nodeAddressBook = Network.nodeAddressBooks.get(networkName);
+        if (nodeAddressBook == null) {
+            throw new IllegalStateException("Unrecognized network name");
+        }
+
+        var nodeAddress = nodeAddressBook.get(nodeAccountId);
+        if (nodeAddress == null) {
+            throw new IllegalStateException("could not find node account ID within address book");
+        }
+
+        this.certHash = new String(nodeAddress.certHash.toByteArray(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (certHash == null) {
+            return;
+        }
+
+        for (var cert : chain) {
+            var digest = new SHA384Digest();
+            var certHash = new byte[digest.getDigestSize()];
+            digest.update(cert.getEncoded(), 0, cert.getEncoded().length);
```

Comment:
Call `cert.getEncoded()` once and reuse as it looks like it clones the byte array internally.

Reply:
I'm also worried if this method is implemented for different ssl implementations. A lot of the implementations I saw throw a not implemented exception. Perhaps we can double check if there's some other appropriate method we can use that's guaranteed to be implemented.